### PR TITLE
Make clue scroll map hint stay within world map borders, and make hint more visible.

### PR DESCRIPTION
**Pull request by critzd**
Change the interaction with the map itself to make it so on full screen the hint stays more visible and within view overall. In addition, change the icon itself to stand out from the overall Runescape map without impacting other clue scroll-map interactions.

Example of this in action:

java_2018-05-22_17-28-09

java_2018-05-22_17-28-29

There is a slight issue with a bit of jumping, but this is done to prevent the icon from moving outside the border, and doesn't impact the user's ability to use the icon or know direction, and only happens when right next to the location itself. Gif below demonstrates this.

### CODE REVIEW

**Comment by Nightfirecat** [comment]
Is the icon still clickable to scroll the map to its location?

**Comment by critzd** [comment]
Still is possible, The second gif makes it look like it closes out but that's just due to clicking on the actual X to close the map. No functionality has changed, just the visual.

**Comment by deathbeam** [comment]
I would at least add some padding to the icon, currently it feels a bit, mashed together.

**Comment by critzd** [comment]
So do you want the box larger that surrounds the clue? Or do you mean add a transparent border around it

**Comment by deathbeam** [comment]
Yeah, slightly larger box, so clue corner is not right at the box corner

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollWorldMapPoint.java [hunk 1]
```diff
@@ -26,15 +26,16 @@
...
 import net.runelite.api.Point;
 import net.runelite.api.coords.WorldPoint;
 import net.runelite.client.ui.overlay.worldmap.WorldMapPoint;
 
 class ClueScrollWorldMapPoint extends WorldMapPoint
 {
 	private static final BufferedImage CLUE_SCROLL_WORLD_IMAGE;
+	private static final BufferedImage CLUE_SCROLL_WORLD_IMAGE_EDGE;
 	private static final Point CLUE_SCROLL_WORLD_IMAGE_POINT;
-
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollWorldMapPoint.java]: I believe there should be a blank line here. You should double check what other static blocks do

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/MapClue.java [hunk 2]
```diff
@@ -77,7 +77,7 @@
 		new MapClue(CLUE_SCROLL_HARD_3520, new WorldPoint(2616, 3077, 0)),
 		new MapClue(CLUE_SCROLL_HARD_3522, new WorldPoint(2488, 3308, 0)),
 		new MapClue(CLUE_SCROLL_HARD_3524, new WorldPoint(2457, 3182, 0), CRATE_18506),
-		new MapClue(CLUE_SCROLL_HARD_3525, new WorldPoint(3026, 3629, 0), CRATE_354),
+		new MapClue(CLUE_SCROLL_HARD_3525, new WorldPoint(3026, 3628, 0), CRATE_354),
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/MapClue.java]: This should be submitted as a separate PR

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlay.java [hunk 3]
```diff
@@ -58,10 +59,13 @@
 	private static final int TOOLTIP_OFFSET_WIDTH = 5;
 	private static final int TOOLTIP_PADDING_HEIGHT = 1;
 	private static final int TOOLTIP_PADDING_WIDTH = 2;
+	private static final int CLUE_HINT_OFFSET = 32;
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlay.java]: WorldMapOverlay is a generic class, it should not contain any clue logic

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollWorldMapPoint.java [hunk 4]
```diff
@@ -69,4 +79,41 @@ public void onEdgeUnsnap()
 		this.setImage(CLUE_SCROLL_WORLD_IMAGE);
 		this.setImagePoint(CLUE_SCROLL_WORLD_IMAGE_POINT);
 	}
+
+	@Override
+	public void edgeSnapImageHandler(boolean north, boolean south, boolean east, boolean west, BufferedImage image)
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollWorldMapPoint.java]: You don't need an entire handler, this could be done much cleaner by changing the default behavior of an edge snapped point to offset draw point such that the entire point image is within the bounds
